lib/pyszn/parser.py

```diff
@@ -97,17 +98,29 @@ def build_parser():
     """
     ParserElement.setDefaultWhitespaceChars(' \t')
     nl = Suppress(LineEnd())
-    number = Word(nums).setParseAction(lambda l, s, t: int(t[0]))
+    inumber = Word(nums).setParseAction(lambda l, s, t: int(t[0]))
+    fnumber = (
```

Comment:
Hi, feel free to ignore me since I don't have any stakes on this lib anymore, and I am just passing by while I unsubscribe from it.

This floating point number is technically incorrect as it doesn't accept negative floats (`-1.0`) and usually floats can also be written as `-2.5e-24` or `3.34E7` so a more correct form would be:
```python
Combine(Optional('-') + Word(nums) + '.' + Word(nums) + Optional('e' | 'E')+ Optional('-') + Optional(Word(nums)))
```
Also, according to IEE-754 floating point standard, `NaN` and +/- `inf` are valid floating point numbers, however those are probably not very useful on attributes...

Reply:
Thanks for your comments @jcrono. We'll stick to floating point numbers that match Python 3 [accepted representation](https://docs.python.org/3/library/functions.html#float).

@eduar-m can you adjust your parser to support these changes? If the parser grows too large, please support at least the case for negative floats.

Reply:
@jcrono Thanks for pointing out the 'e' and 'E' usage is a nice addition to the floats.
@ocelotl PR back to you I added support for negatives too.